test/src/folding/foldable_block_test.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:flutter_code_editor/src/folding/foldable_block.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'parsers/test_executor.dart';
+
+void main() {
+  group('Foldable block test', () {
+    test('Join intersected blocks in empty list will not throw exception', () {
+      const blocks = <FoldableBlock>[];
+
+      final actual = blocks.deepCopy()..joinIntersecting();
+
+      expect(actual, blocks);
+    });
+
+    test('Join blocks in list with one block will not throw exception', () {
+      const blocks = [
+        FoldableBlock(startLine: 0, endLine: 1, type: FBT.braces),
+      ];
+
+      final actual = blocks.deepCopy()..joinIntersecting();
+
+      expect(actual, blocks);
+    });
+
+    test('Join intersecting blocks in list with two intersecting blocks', () {
+      const blocks = [
+        FoldableBlock(startLine: 0, endLine: 1, type: FBT.braces),
+        FoldableBlock(startLine: 1, endLine: 2, type: FBT.braces),
+      ];
+      const matcher = [
+        FoldableBlock(startLine: 0, endLine: 2, type: FBT.union),
+      ];
+
+      final actual = blocks.deepCopy()..joinIntersecting();
+
+      expect(actual, matcher);
+    });
+
+    test('Join intersecting blocks in list with two not intersecting blocks',
```

Comment:
Do we need this test if we have the one after it?

Reply:
We do, because it's edge case. Two items are minimal available count for intersecting

Reply:
Does it work differently internally? If no, then my experience is that less tests on similar inputs is preferred. This is because if we change the API we will have to refactor tests which is more tedious than refactoring code.

Reply:
Fixed tests